### PR TITLE
[Merged by Bors] - [Merged by Bors] - feat: forward-port #18148

### PR DESCRIPTION
**Pull request by 101damnations**
---

* [`algebra.category.Mon.basic`@`4125b9adf2e268d1cf438092d690a78f7c664743`..`0caf3701139ef2e69c215717665361cda205a90b`](https://leanprover-community.github.io/mathlib-port-status/file/algebra/category/Mon/basic?range=4125b9adf2e268d1cf438092d690a78f7c664743..0caf3701139ef2e69c215717665361cda205a90b)
* [`category_theory.closed.monoidal`@`c0e00a871b9f6d3aca7c10fb3abdc8720a2c5313`..`0caf3701139ef2e69c215717665361cda205a90b`](https://leanprover-community.github.io/mathlib-port-status/file/category_theory/closed/monoidal?range=c0e00a871b9f6d3aca7c10fb3abdc8720a2c5313..0caf3701139ef2e69c215717665361cda205a90b)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by kbuzzard** [comment]
This seems to faithfully forward-port the relevant parts of the mathlib3 PR 18148. Why is `Mathlib/Algebra/Hom/Group.lean` mentioned on the "Files Changed" tab?

**Comment by semorrison** [comment]
> This PR seems to faithfully forward-port the relevant parts of the mathlib3 PR 18148. Why is `Mathlib/Algebra/Hom/Group.lean` mentioned on the "Files Changed" tab?

It no longer is.

File: Mathlib/CategoryTheory/Closed/Monoidal.lean [hunk 1]
```diff
@@ -313,6 +313,39 @@ noncomputable def ofEquiv (F : MonoidalFunctor C D) [IsEquivalence F.toFunctor]
         exact Adjunction.leftAdjointOfNatIso i }
 #align category_theory.monoidal_closed.of_equiv CategoryTheory.MonoidalClosed.ofEquiv
 
+/- ./././Mathport/Syntax/Translate/Expr.lean:177:8: unsupported: ambiguous notation -/
+/- ./././Mathport/Syntax/Translate/Expr.lean:177:8: unsupported: ambiguous notation -/
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Closed/Monoidal.lean:317]:
```suggestion
```

File: Mathlib/CategoryTheory/Closed/Monoidal.lean [hunk 2]
```diff
@@ -313,6 +313,39 @@ noncomputable def ofEquiv (F : MonoidalFunctor C D) [IsEquivalence F.toFunctor]
...
+    [MonoidalClosed D] {X Y Z : C} (f : X ⊗ Y ⟶ Z) :
+    @MonoidalClosed.curry _ _ _ _ _ _ ((MonoidalClosed.ofEquiv F).1 _) f =
+      (F.1.1.adjunction.homEquiv Y ((ihom _).obj _))
+        (MonoidalClosed.curry
+          (F.1.1.inv.adjunction.homEquiv (F.1.1.obj X ⊗ F.1.1.obj Y) Z
+            ((compInvIso (F.commTensorLeft X)).hom.app Y ≫ f))) :=
+  rfl
+#align category_theory.monoidal_closed.of_equiv_curry_def CategoryTheory.MonoidalClosed.ofEquiv_curry_def
+
+/- ./././Mathport/Syntax/Translate/Expr.lean:177:8: unsupported: ambiguous notation -/
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Closed/Monoidal.lean:333]:
```suggestion
```

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**